types/openseadragon/index.d.ts

```diff
@@ -0,0 +1,1468 @@
+
+export function getElement(element: string | Element): Element;
+
+export function getElementOffset(element: Element | string): Point;
+
+export function getElementPosition(element: Element | string): Point;
+
+export function getElementSize(element: Element | string): Point;
+
+export function getElementStyle(element: Element | string): any; // CSSStyle?
+
+export function getMousePosition(event?: OSDEvent<any>): Point;
+
+export function getPageScroll(): Point;
+
+export function getString(property: string): string;
+
+export function getUrlParameter(key: string): string;
+
+export function getWindowSize(): Point;
+
+export function imageFormatSupported(extension?: string): boolean;
+
+export function indexOf(
+    array: any[],
+    searchElement: object,
+    fromIndex?: number
+): number;
+
+// (missing jquery functions)
+
+export function makeAjaxRequest(options: {
+    url: string;
+    success: (obj: object) => void;
+    error: (obj: object) => void;
+    headers: object;
+    responseType: string;
+    withCredentials?: boolean;
+}): XMLHttpRequest;
+
+export function makeCenteredNode(element: Element | string): Element;
+
+export function makeNeutralElement(tagName: string): Element;
+
+export function makeTransparentImage(src: string): Element;
+
+export function now(): number;
+
+export function parseJSON(string: string): object;
+
+export function parseXml(string: string): Document;
+
+export function pointInElement(
+    element: Element | string,
+    point: Point
+): boolean;
+
+export function positiveModulo(number: number, modulo: number): number;
+
+export function removeClass(element: Element | string, className: string): void;
+
+export function removeEvent(
+    element: Element | string,
+    eventName: string,
+    handler: EventHandler<any>,
+    useCapture?: boolean
+): void;
+
+export function setElementOpacity(
+    element: Element | string,
+    opacity: number,
+    usesAlpha?: boolean
+): void;
+
+export function setElementTouchActionNone(element: Element | string): void;
+
+export function setPageScroll(point: Point): void;
+
+export function setString(property: string, value: any): void;
+
+export function stopEvent(event?: OSDEvent<any>): void;
+
+export interface GestureSettings {
+    scrollToZoom?: boolean;
+    clickToZoom?: boolean;
+    dblClickToZoom?: boolean;
+    pinchToZoom?: boolean;
+    flickEnabled?: boolean;
+    flickMinSpeed?: number;
+    flickMomentum?: number;
+}
+
+interface NavImagesValues {
+    REST: string;
+    GROUP: string;
+    HOVER: string;
+    DOWN: string;
+}
+export interface NavImages {
+    zoomIn: NavImagesValues;
+    zoomOut: NavImagesValues;
+    home: NavImagesValues;
+    fullpage: NavImagesValues;
+    rotateleft: NavImagesValues;
+    rotateright: NavImagesValues;
+    flip: NavImagesValues;
+    previous: NavImagesValues;
+    next: NavImagesValues;
+}
+
+export interface Options {
+    id?: string;
+    element?: HTMLElement;
+    tileSources?:
```

Comment:
Should this definition for `TileSource` shape contains a `type` property somewhere? Event optional?
There are sampels of code snips that contains type:
https://openseadragon.github.io/examples/tilesource-image/
, the project sample code does not compile due to this:
```ts
const viewer3 = OpenSeadragon({
    id: "openseadragon",
    prefixUrl: "openseadragon/images/",
    showNavigator: false,
    tileSources: {
      type: 'legacy-image-pyramid',
      levels:[{
          url: '2003rosen1799/0001q.jpg',
          height: 889,
          width:  600
      },{
          url: '2003rosen1799/0001r.jpg',
          height: 2201,
          width:  1485
      },{
          url: '2003rosen1799/0001v.jpg',
          height: 4402,
          width:  2970
      }]
    }
  });
```

Reply:
:+1: you're right, I also added the example to the tests